src/modules/computerVision/gaborFilter/gaborFilter.tsx

```diff
@@ -0,0 +1,184 @@
+    const [kernelSize, setKernelSize] = useState<number>(5);
+    const [omega, setOmega] = useState<number>(1);
+    const [theta, setTheta] = useState<number>(0);
+    const [K, setK] = useState<number>(3.14);
+
+    const changeOmega = (e: any) => setOmega(parseFloat(e.target.value));
+    const changeKernelSize = (e: any) => setKernelSize(parseInt(e.target.value));
+    const changeTheta = (e: any) => setTheta(parseFloat(e.target.value));
+    const changeK = (e: any) => setK(parseFloat(e.target.value));
+
+    const invalidSize = (kernelSize < 3 || kernelSize > 101)
+    const invalidConfig = invalidSize;
+    return (
+        <div>
+            <div className="font-bold m-3">
+                Omega
+                <input className="mx-2 w-64"
+                    type="range" min="0" max="10"
+                    value={omega} onChange={(e) => changeOmega(e)} />
+                <input className="number-input"
+                    type="number" min="0" max="10"
+                    value={omega} onChange={(e) => changeOmega(e)} />
+            </div>
+            <div className="font-bold m-3 h-10">
+                Kernel Size
+                <input className="mx-2 w-64"
+                    type="range" min="3" max="101" step={2}
+                    value={kernelSize} onChange={(e) => changeKernelSize(e)} />
+                <input className="number-input"
+                    type="number" min="3" max="101" step={2}
+                    value={kernelSize} onChange={(e) => changeKernelSize(e)} />
+                <div className="font-light italic text-sm">
+                    { invalidSize ? 'Enter an odd kernel size, between 3 and 101' : ''}
+                </div>
+            </div>
+            <div className="font-bold m-3 h-10">
+               Theta
+               <input className="mx-2 w-64"
+                     // Math.PI/4 ~= 0.79
+                    type="range" min="0" max="6.28" step={0.785}
+                    value={theta} onChange={(e) => changeTheta(e)} />
+               <input className="number-input"
+                    type="number" min="0" max="6.28" step={0.785}
+                    value={theta} onChange={(e) => changeTheta(e)} />
+            </div>
+            <div className="font-bold m-3 h-10">
+               K
+               <input className="mx-2 w-64"
+                     // Math.PI/4 ~= 0.79
+                    type="range" min="0" max="12.56" step={0.785}
+                    value={K} onChange={(e) => changeK(e)} />
+               <input className="number-input"
+                    type="number" min="0" max="12.56" step={0.785}
+                    value={K} onChange={(e) => changeK(e)} />
+            </div>
+            <button className="basic-button" disabled={invalidConfig} onClick={e => props.onConfig(kernelSize, omega, theta, K)}>
+                Generate Kernel
+            </button>
+        </div>
+    );
+}
+
+function getBg(val: number, kernel?: number[]) {
```

Comment:
This is the same as `getBg()` in the Gaussian filter, right? we can probably just add the function to the exports for Gaussian, so it doesn't need to be copied in this file

Reply:
yes that sounds good. Im pretty sure I didnt change that function at all from your version in gaussian blur